### PR TITLE
fix: remove `module_ctx.extension_metadata`

### PR DESCRIPTION
**Pull request by fmeum**
In the current Bzlmod setup, all repositories generated by the `graalvm` extension may or may not be used by users and whether they should be can't be predicted based on the given tags: Users that want to generate toolchains need the `<name>_toolchains` repo, users that rely on SDK files need the `<name>` repo.

For now, remove the use of `module_ctx.extension_metadata` so that no warning is shown when users register toolchains in `<name>_toolchains`. In the future, we should improve the way toolchains are registered with Bzlmod by automatically registering a toolchain configured via tags.

### CODE REVIEW

**Comment by fmeum** [comment]
@sgammon Could you create a 0.10.2 release after merging this? That should unblock the work on turbine while we look into further improvements such as https://github.com/sgammon/rules_graalvm/pull/124.

**Review by sgammon — APPROVED** [review]

**Comment by sgammon** [comment]
sounds good 👍🏻

**Pull request merged**
